### PR TITLE
PORTALS-2078

### PR DESCRIPTION
**Pull request by jay-hodgson**
If there's a file upload error while gaining access, show it

### CODE REVIEW

File: src/lib/containers/access_requirement_list/managedACTAccess/RequestDataAccessStep2.tsx [hunk 1]
```diff
@@ -450,45 +452,55 @@ const RequestDataAccessStep2: React.FC<RequestDataAccessStep2Props> = props => {
   }
 
   const uploadCallback = (data: UploadCallbackResp) => {
-    if (data.context === 'attachments') {
```
Comment (jay-hodgson) [inline src/lib/containers/access_requirement_list/managedACTAccess/RequestDataAccessStep2.tsx:453]: This is a terrible diff, I wrapped the existing code with:
if (data.resp && data.success) {
...
}

File: src/lib/containers/access_requirement_list/managedACTAccess/RequestDataAccessStep2.tsx [hunk 2]
```diff
@@ -450,45 +452,55 @@ const RequestDataAccessStep2: React.FC<RequestDataAccessStep2Props> = props => {
...
+        if (data.context === 'irbFileHandleId') {
+          setIRB(prevState => {
+            return Object.assign({}, prevState, {
+              fileName: uploadResponse.fileName,
+              fileHandleId: uploadResponse.fileHandleId,
+            })
+          })
+        }
       }
+    } else if (!data.success && data.error) {
```
Comment (jay-hodgson) [inline src/lib/containers/access_requirement_list/managedACTAccess/RequestDataAccessStep2.tsx:497]: And this is the new code.

File: src/lib/containers/access_requirement_list/managedACTAccess/RequestDataAccessStep2.tsx [hunk 3]
```diff
@@ -59,8 +59,10 @@ export type DataAccessDocs =
   | undefined
 
 export type UploadCallbackResp = {
-  resp: FileUploadComplete
+  resp?: FileUploadComplete
```
Comment (jay-hodgson) [inline src/lib/containers/access_requirement_list/managedACTAccess/RequestDataAccessStep2.tsx:62]: response is not set if there was an error

**Comment by jay-hodgson** [comment]
On error after change: Screen Shot 2021-12-14 at 3 04 41 PM

**Review by nickgros — APPROVED** [review]

**Pull request merged**